drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,253 @@
+
+With the version 3 transaction format, the first four bytes of a serialized transaction, the 32-bit header, are made up of two fields as shown in the able above:
+
+* 1 bit Overwinter flag, must be set
+* 31 bit unsigned int for the Version
+
+Pre-Overwinter parsers will deserialize these four bytes as a 32-bit signed integer.  With two's complement integers, the most significant bit indicates whether an integer is positive or negative.  With the Overwinter flag set, the transaction version will be negative, resulting in pre-Overwinter parsers rejecting the transaction as invalid.  This provides transaction replay protection between per-Overwinter and Overwinter software.
+
+Consider the following example of a serialized version 3 transaction.
+
+Pre-Overwinter parser:
+
+* data begins with little-endian byte sequence: [0x03, 0x00, 0x00, 0x80]
+* deserialized as 32-bit signed integer
+
+  * with hexadecimal value of 0x80000003 (most significant bit is set)
+  * decimal value of -2147483645
+
+Legacy parsers will expect the version to be a positive value, such as 1 or 2, and will thus reject the Overwinter transaction as invalid.
+
+Overwinter parser:
+
+- data begins with little-endian byte sequence: [0x03, 0x00, 0x00, 0x80]
+- deserialized as 32-bit unsigned integer
+  
+  - with binary value of 10000000000000000000000000000011
+- decomposed into two fields  
+  
+  - overwinter flag (bit 31) is set
+  - version (bits 30 - bit 0) have a decimal value of 3
+
+Overwinter parsers will accept the transaction as valid as the most significant bit of the header has been set.  By masking off (unsetting) the most significant bit, the parser can retrieve the transaction version number::
+
+    0x80000003 & 0x7FFFFFFFF = 0x00000003 = 3
+
+Version Group Id
+----------------
+
+The version group id is a random and unique identifier assigned to a transaction format version or a group of soft-forking transaction format versions.  The version group id helps nodes disambiguate between branches using the same version number.
+
+That is, it prevents a client on one branch of the network from attempting to parse transactions intended for another branch, in the situation where the transactions share the same format version number but are actually specified differently.  For example, Zcash and Zclone both define their own custom v3 transaction formats, but each will have its own unique version group id, so that they can reject v3 transactions with unknown version group ids.
+
+The combination of transaction version and version grouph id, `nVersion || nVersionGroupId` uniquely defines the transaction format, thus enabling parsers to reject transactions from outside the client's chain which cannot be parsed.  This helps provide users with a layer of replay protection at the parser level.  Full replay protection is defined in the Overwinter Transaction Signature Verification scheme [#zip-0???]_.
```

Comment:
`grouph -> group`

Reply:
```
`nVersion || nVersionGroupId`, uniquely
```

Reply:
I'm not sure it's accurate to say it's a layer of replay protection, if the unparseable transaction format would never be valid on the chain anyway...